src/protocol/tags.rs

```diff
@@ -17,14 +17,28 @@ impl std::ops::Deref for Tags {
 impl FromValue for Tags {
     fn from_value(value: Annotated<Value>) -> Annotated<Self> {
         type TagTuple = (Annotated<LenientString>, Annotated<LenientString>);
+        fn check_chars(x: String) -> Result<String, Annotated<String>> {
+            if x.contains('\n') {
+                Err(Annotated::from_error(
```

Comment:
_nit_: That could go in the store processor (or some processor) to simplify `FromValue`. It is a kind of validation.

Reply:
The entire tags impl is annoying. I can't use any annotations here otherwise I'd have used `match_regex`.

Eventually I want to rewrite this to use match_regex and write a separate processor that processes those kind of annotations.